yaksh/views.py

```diff
@@ -605,7 +605,7 @@ def course_detail(request, course_id):
     user = request.user
     if not is_moderator(user):
         raise Http404('You are not allowed to view this page')
-    course = get_object_or_404(Course, creator=user, pk=course_id)
+    course = Course.objects.filter(Q(creator=user)|Q(teachers=user), pk=course_id).first()
```

Comment:
If user is not a moderator an exception is raised. Similarly an exception should be raised if the user is not allotted to the requested course.
In this case, I can copy-paste the url. Though course will be None, I can access the page.
So I can view other moderator quizzes as well.